app/next-client-app/components/data-table/DataTablePagination.tsx

```diff
@@ -34,18 +35,19 @@ export function DataTablePagination<TData>({
   const numberOfPages = Math.max(Math.ceil(count / pageSize), 1);
 
   // Notice:
-  // The condition below avoids the parent page to break when users change the page size to a larger number which making the app couldn't find the data with the according p and page_size params in query URL.
-  // If the user actually does this "rarely-happened" action, they will see a "No results" page, then be pushed back to the first page with the page size they have chosen. An Error about Router in the console will also appear.
-  // The more effective fix can be adding some logics in the API endpoint getting the data, but it also means that we need to change many other API endpoints as well. So it may not be desirable.
+  // The condition below avoids the parent page to break when users change the page size to a larger number
+  // which making the app couldn't find the data with the according p and page_size params in query URL.
+  // If the user actually does this "rarely-happened" action, they will see a "No results" page,
+  // then be pushed back to the first page with the page size they have chosen.
+  // The more effective fix can be adding some logics in the API endpoint getting the data,
+  // but it also means that we need to change many other API endpoints as well. So it may not be desirable.
 
-  if (currentPage > numberOfPages) {
-    navigateWithSearchParam(
-      "p",
-      Math.ceil(count / pageSize),
-      router,
-      searchParams
-    );
-  }
+  // Using useEffect here to prevent an error about Router on the console
+  useEffect(() => {
```

Comment:
Testing this on dev - there seems to be a Next server side error when navigating to the last page of Scan Report Values, then increasing the page size (replicating the behaviour this is trying to solve).
Additionally, on the Data Sets page, doing the same thing produced a `NaN`, and the pages could be infinitely navigated.

<img width="528" alt="Screenshot 2024-06-06 at 19 14 58" src="https://github.com/Health-Informatics-UoN/Carrot-Mapper/assets/1127507/434f3a80-a485-4dea-ad7b-6ef0a68c11e8">

If this isn't a quick fix, let's remove this `useEffect` for now and keep the existing behaviour.